### ISSUE TITLE
Change all internal value handling to wei (not eth)

### ISSUE THREAD

**Issue by wanderingstan** [OriginProtocol/demo-dapp#185]
Currently, we store listing prices in ETH throughout the the DApp (and in origin-js) This is bad practice for monetary values. (And it's my fault for making this bad design choice last year. 🤦‍♀️)

We should convert to always representing prices in Wei (or fundamental unit of tokens, when that comes) and only convert to ETH at render time. 

This will also prepare us for handling ERC20 tokens with arbitrary number of decimals.

**Comment by DanielVF** [comment]
I'm thinking we'll be passing and returning these numbers as strings from the origin.js API. This lets us use a json compatible type. It's easy to create a BigNumber from a string, which you would be doing anyway to use the value.

**Comment by wanderingstan** [comment]
@DanielVF is that because BigNumbers are so hard to deal with, and always display strangely? 

I'm with you in spirit, but worry about introducing a third representation format. (js Number, BigNumber, string of number) 

It seems the "right" way would be to use BigNumber in the DApp, but I agree it's a pain. Anyone else have opinions on this?

**Comment by nick** [comment]
Agree we should represent everything in wei and pass strings back from origin-js. web3.utils.fromWei doesn't work with numbers, it will throw `Please pass numbers as strings or BigNumber objects to avoid precision errors.`

**Comment by wanderingstan** [comment]
If `web3.utils` accepts strings (along with BigNumbers), then thats's good enough for me! I retract any hesitation about using them for origin-js/DApp.

**Comment by gustavoguimaraes** [comment]
Hi all,

Is this issue still applicable?  I see some commits that changes the prices from wei to eth for rendering purposes only in the codebase.  This issue shows as a `good first issue` but I cannot tell if it is still relevant given that it is from May.

**Comment by wanderingstan** [comment]
Hi @gustavoguimaraes thanks for bringing this up. I believe things have migrated a bit in the codebase and now we actually are storing things in raw eth in many places. This is a discussion we need to re-visit.

So yes, don't do any work on this until we figure out what path we want to take.

**Comment by franckc** [comment]
I don't think this would require any change to the  JSON schemas (since there is no validation at the schema level of the monetary unit being used). But logic code in origin-js would be impacted. Since with the GraphQL rewrite we planning to replace origin-js by new code, this could be a good opportunity to clean this up ?

I could imagine having a "Money" class at the business logic layer that gets constructed off data from IPFS. That constructor should be smart enough to handle legacy IPFS data for old listings/offers that were not using natural units. Then the Money object would be what gets used throughout the DApp. And it would provide helper methods like basic math operations, getting token symbol, unit / natural unit conversion and representation, etc...

**Comment by micahalcorn** [comment]
Just leaving a note here that @crazybuster and I are having trouble dealing with BigNumbers 😕 

![say-what](https://user-images.githubusercontent.com/273937/50994080-24859c80-14d0-11e9-8d48-9d0057528568.png)

[According to the docs](https://web3js.readthedocs.io/en/1.0/web3-utils.html#tobn)...

**toBN**
_Will safely convert any given value (including BigNumber.js instances) into a BN.js instance, for handling big numbers in JavaScript._

**isBN**
_Checks if a given value is a BN.js instance._

**Comment by DanielVF** [comment]
@micahalcorn Ever find out what was the deal with isBN?

**Comment by micahalcorn** [comment]
Nope 😕

**Comment by DanielVF** [comment]
Well, this is fun. It works fine in dapp2.

![image](https://user-images.githubusercontent.com/837/52146002-8bf6be00-2630-11e9-8f98-b179cbd396e4.png)

The web3 code for `utils.isBN` hasn't changed in a year, nor has the code for web3 code `toBN`.  The BN dependencies used by `web.utils` have not have their version numbers changed in two years.

It looks like the difference is in what is returned by "toBN". I wonder if there's some kind of webpack weirdness going on with DAPP1. 

![image](https://user-images.githubusercontent.com/837/52147100-b138fb80-2633-11e9-8bbb-4c8fe4ac7996.png)

Sadly, I've got to run for now before chasing this all the way down to the ground.

**Comment by micahalcorn** [comment]
Still relevant per @nick 🙂
